drivers/I2C.h

```diff
@@ -74,7 +74,7 @@ namespace mbed {
  *         i2c.read( addr8bit, cmd, 2);
  *
  *         float tmp = (float((cmd[0]<<8)|cmd[1]) / 256.0);
- *         printf("Temp = %.2f\n", tmp);
+ *         mbed_printf("Temp = %.2f\n", tmp);
```

Comment:
Floating point support for minimal-printf is disabled by default.

Reply:
Then the minimal-printf README is wrong:
```
* %f: floating point (enabled by default).
* %F: floating point (enabled by default, treated as %f).
* %g: floating point (enabled by default, treated as %f).
* %G: floating point (enabled by default, treated as %f).
```

Reply:
I'll add it to things to consider in this PR.

Reply:
Yes it is wrong: https://github.com/ARMmbed/mbed-os/blob/master/platform/mbed_lib.json#L146